vignettes/symptom-surveys.Rmd

```diff
@@ -0,0 +1,672 @@
+
+## Forecasting Code
+
+The code below marches the forecast date $t_0$ forward, one day at a time for the nine forecasting dates for which the four models can all be fit (May 6, 2020 to May 14, 2020). Then, it fits the models, makes predictions 7 and 14 days ahead (as permissible by the data), and records the errors. 
+
+There are a number of benefits to using `epipredict` over writing the code from scratch to fit and predict under each of the models. First, we do not have to reformat the data for input into a model or concern ourselves with its unique interface. We instead work under unifying interface to streamline the modelling process. Second, we avoid having to write our own function to append shift values (leads or lags). This is done for us under-the-hood in the `arx_forecaster()` function. You can see this in the forecaster output by inspecting the `step_epi_lag()` and `step_epi_ahead()` pre-processing steps in the `epi_workflow`. Third, we only need one for loop for the forecast dates (and not a second loop for the different aheads) as we can easily use `map()` with the `arx_forecaster()` over the different ahead values, as we’ve done before.
+
+However, there are some trade-offs to bear in mind. For instance, since we are using a canned arx forecaster, we are not able to easily modify and add steps such as that for signal transformations to the pre-processing (this is pre-specified as part of using a canned forecaster). If we were to code-up our own forecaster under the `epipredict` framework, we could easily add steps to re-scale and transform the signals to our `epi_recipe`. This would make the code more succinct and self-contained.
+
+```{r, message = FALSE, warning = FALSE}
+library(covidcast)
+library(dplyr)
+library(purrr)
+library(epipredict)
+library(recipes)
+
+case_num <- 200
+as_of_date <- "2020-05-14"
+geo_values <- covidcast_signal(
+  "jhu-csse", "confirmed_cumulative_num",
+  "2020-05-14", "2020-05-14"
+) %>%
+  filter(value >= case_num) %>%
+  pull(geo_value)
+
+# Fetch county-level Google and Facebook % CLI-in-community signals, and JHU
+# confirmed case incidence proportion
+start_day <- "2020-04-11"
+end_day <- "2020-09-01"
+
+g <- covidcast_signal("google-survey", "smoothed_cli") %>%
+  filter(geo_value %in% geo_values) %>%
+  select(geo_value, time_value, value)
+f <- covidcast_signal(
+  "fb-survey", "smoothed_hh_cmnty_cli",
+  start_day, end_day
+) %>%
+  filter(geo_value %in% geo_values) %>%
+  select(geo_value, time_value, value)
+c <- covidcast_signal(
```

Comment:
Ok. I have changed these names to be more descriptive. I had them as g, f, and c because those were the names used n the [blog post](https://delphi.cmu.edu/blog/2020/09/21/can-symptoms-surveys-improve-covid-19-forecasts/) that this chapter is based on.